### PR TITLE
Fixes #34160 - Improve fact_values api performance

### PR DESCRIPTION
**Pull request by tbrisker**
Previously, the fact_values api called loaded a lot of un-needed data
from the database and used a sub-optimal way of converting the data to a
hash for display.
This commit changes the implementations so that we only load the
required columns from the database and generate the hash in a much more
efficient way. It also moves the build_facts_hash method from FactValue
model to the controller, which is its only consumer.

Some benchmarks:
Before, loading 20 facts (default per_page setting):
Completed 200 OK in 52ms (Views: 2.6ms | ActiveRecord: 15.5ms | Allocations: 35603)
Before, loading 1000 facts:
Completed 200 OK in 1181ms (Views: 2.2ms | ActiveRecord: 22.8ms | Allocations: 1390471)
Before, loading 10000 facts:
Completed 200 OK in 14581ms (Views: 7.4ms | ActiveRecord: 75.6ms | Allocations: 22283568)

After, loading 20 facts:
Completed 200 OK in 37ms (Views: 3.8ms | ActiveRecord: 18.3ms | Allocations: 15549)
After, loading 1000 facts:
Completed 200 OK in 53ms (Views: 2.4ms | ActiveRecord: 16.0ms | Allocations: 47083)
After, loading 10000 facts:
Completed 200 OK in 313ms (Views: 5.8ms | ActiveRecord: 38.7ms | Allocations: 433106)

### CODE REVIEW

File: test/controllers/api/v2/fact_values_controller_test.rb [hunk 1]
```diff
@@ -27,15 +27,15 @@ def setup
     get :index, params: { :host_id => @host.name }
     assert_response :success
     fact_values   = ActiveSupport::JSON.decode(@response.body)['results']
-    expected_hash = FactValue.build_facts_hash(FactValue.where(:host_id => @host.id))
+    expected_hash = {@host.name => {"kernelversion" => "2.6.9"}}
```
Comment (ekohl) [inline test/controllers/api/v2/fact_values_controller_test.rb:30]: Big :+1: for making this simple. This you can read as a human.

**Review by ekohl — APPROVED** [review]
> Some benchmarks:
Recently I came across https://github.com/rails/rails-perftest and that makes it easy to write those benchmarks on actual code. I think it would be worth investigating if we can use that.

The only thing I don't know is how we're going to compare it between versions.

File: test/controllers/api/v2/fact_values_controller_test.rb (hunk 1 above)
Comment (tbrisker) [reply test/controllers/api/v2/fact_values_controller_test.rb:30]: It is also now testing that the response is actually in the format we expect it to be and not that the controller calls the method correctly.

Comment (ekohl) [reply test/controllers/api/v2/fact_values_controller_test.rb:30]: Indeed. `FactValue.build_facts_hash` could have returned `nil` and still passed this test.

**Pull request merged**